### ISSUE TITLE
L4_Profile: unable to set syn-cookie-enable and syn-cookie-whitelist

### ISSUE THREAD

**Issue by manhac** [F5Networks/f5-appsvcs-extension#330]
Using AS3 you can set up a custom L4_Profile, like if you need to enable loose-initialization (for example for IP forwarding virtual servers: https://support.f5.com/csp/article/K7595):
```json
{
  "class": "ADC",
  "schemaVersion": "3.0.0",
  "fastl4_profile_test": {
    "class": "Tenant",
    "fastl4_profile_test": {
      "class": "Application",
      "template": "generic",
      "fastL4_forwardingserver": {
        "class": "L4_Profile",
        "idleTimeout": 300,
        "looseClose": true,
        "looseInitialization": true,
        "resetOnTimeout": false
      }
    }
  }
}
```

Which creates the following configuration:
```
ltm profile fastl4 fastl4_profile_test/fastl4_profile_test/fastL4_forwardingserver {
    app-service none
    defaults-from Common/fastL4
    idle-timeout 300
    loose-close enabled
    loose-initialization enabled
    mss-override 0
    reset-on-timeout disabled
    tcp-close-timeout 5
    tcp-handshake-timeout 5
}
```

However, it is not possible to configure parameters for SYN cookies. So, if you are doing a config verification (tmsh load /sys config verify) or if you are visiting the TMUI configuration page of the newly created L4_Profile, the system complains: "Loose Initiation and SYN Challenge Handling are both currently enabled. These two features are mutually incompatible and may cause crashing. For more information, see https://support.f5.com/csp/article/K99545311"

To get rid of the warning message (and possibly to avoid crashes), you can set "SYN Challenge Handling" to "Disable Challenges" in the TMUI L4_Profile configuration page of the profile (https://support.f5.com/csp/article/K74451051). This leads to the following configuration:
<pre>
ltm profile fastl4 fastl4_profile_test/fastl4_profile_test/fastL4_forwardingserver {
    app-service none
    defaults-from Common/fastL4
    idle-timeout 300
    loose-close enabled
    loose-initialization enabled
    mss-override 0
    reset-on-timeout disabled
    <b>syn-cookie-enable disabled</b>
    <b>syn-cookie-whitelist disabled</b>
    tcp-close-timeout 5
    tcp-handshake-timeout 5
}
</pre>

It would be beneficial, if there could be a **synCookieEnable** and **synCookieWhitelist** option for the L4_Profile -- for the TCP_Profile they are already available.

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added this to our internal product backlog as AUTOTOOL-1895.

**Comment by melkypie** [comment]
Is there any progress on this? Would be great if the L4_Profile could get some love as it only has a handful of supported options, and this one is pretty crucial to some.

**Comment by mgaliulin** [comment]
Hi, any progress? I need to implement LooseInitiation functionality. If you can't add  switching on/off of syn-cookie-enable, may be you can just switch it off if LooseInitiation is set to true? see https://support.f5.com/csp/article/K99545311 for details.

**Comment by dstokesf5** [comment]
@mgaliulin This issue is scheduled to be worked on very soon and should be available in the 3.31.0 release.

**Comment by dstokesf5** [comment]
This was resolved with the 3.31.0 release of AS3.
